library/swift/src/EnvoyClientBuilder.swift

```diff
@@ -20,6 +20,7 @@ public final class EnvoyClientBuilder: NSObject {
   private var statsFlushSeconds: UInt32 = 60
   private var appVersion: String = "unspecified"
   private var appId: String = "unspecified"
+  private var virtualClusters: String = "[]"
```

Comment:
I'm a little concerned that even though this is temporary, consumers can easily incorrectly add clusters. What do you think about doing something like this:

```
private var virtualClusters = [String]()

  func addVirtualCluster(_ virtualCluster: String) -> EnvoyClientBuilder {
    self.virtualClusters.append(virtualCluster)
    return self
  }
```

Then later you can pass it down as `"[\(self.virtualClusters.joined(separator: ",")]"`

WDYT?

Reply:
I'm not sure your proposal gives much additional protection. In general I think the approach I would like to take here is to actually work with the structured config in the bridge/native layer, and expose ergonomic APIs up to the platform. So for instance the API at the platform layer would not be `addVirtualClusters` which is an implementation detail IMO, but rather `addPathSegmentation` (or something like that). Because what we are really trying to achieve here is higher dimension observability.

Reply:
@junr03  @rebello95 I want to make sure we're in alignment for our objective here. This is all temporary, right? I think we should be more willing to go with what we have now to see how the data looks in the next few days and make adjustments after our beta cut.

What do you think?

Reply:
Yes, 100% agree that this is temporary. Just seems like the suggestion would provide a _little more resiliency_ in the mean time, but this is fine

Reply:
yeah, this approach is temporary.